### ISSUE TITLE
drawing multiedges especially with labels does not work

### ISSUE THREAD

**Issue by shivonkar** [networkx/networkx#5609]
### Current Behavior

The dictionary returned by nx.get_edge_attributes has the structure (source, dest, enum):attr, where the third field just enumerates the occurrences of each edge. This third field is necessary because keys have to be unique in the dictionary. However this will imply that it cannot be used in nx.draw_networkx_edge_labels, because it expects a (source, dest):attr structured dict.

### Expected Behavior

Display label for each edge

### Steps to Reproduce

above mentioned 

### Environment

Python version: 3.8
NetworkX version: latest

**Comment by RobinSobczyk** [comment]
Had the exact same problem ! There seem to be workarounds by exporting to a dot file and using pydot/pygraphviz. However, I find matplotlib more convenient for lots of things, like widgets that are useful for teaching/interactive notebooks etc.

**Comment by shivonkar** [comment]
Hi,The workaround may not work in production where we do not have access.
Hope, it will be fixed soon
Thank you, Shiv, 91-9108282540 

    On Friday, 6 May, 2022, 10:03:32 pm IST, RobinSobczyk ***@***.***> wrote:  
 
 


Had the exact same problem ! There seem to be workarounds by exporting to a dot file and using pydot/pygraphviz. However, I find matplotlib more convenient for lots of things, like widgets that are useful for teaching/interactive notebooks etc.

**Comment by RobinSobczyk** [comment]
That's true. I enquired a bit and found that it was not that easy to draw parallel edges with labels. Some code for multiedges (or at least a starting point for it) can be found here : https://stackoverflow.com/questions/22785849/drawing-multiple-edges-between-two-nodes-with-networkx
Though, there are still problems to plot the labels, as many user would like it to be plotted at the middle of the edge. Matplotlib does not provide a nice function to interpolate the path (as you can see here : https://matplotlib.org/stable/api/path_api.html#matplotlib.path.Path.interpolated , it currently only supports well straight lines), and thus finding the middle is not that easy within matplotlib.

However, I can think of two things :
- matplotlib uses only straight lines, quadratic and cubic Bezier curves, and as those are polynomials of low degree in disguise, there may be a closed form to compute the length of such curves. With this closed form, you can then compute the whole length (intergrate path for t=0 to 1), and you'll still have to figure which is the t_mid such that integrating the path from 0 to t_mid results in half the complete length, and then compute the resulting point for t_mid. Depending on the closed form complexity (provided it exists), determining t_mid can go from quite easy to completely unfeasible.
- there are still nice interpolation features like the ones in scipy, which will perform way better than the current matplotlib function. With a good enough linear interpolation, you can find easily an approximation of the middle of the length by just summing the segment lengths, which are easy to computes. This kind of approximation can then be furthermore refined by  using integration tools that are more precise than a linear interpolation.

Some links that provide sources that can be useful :
https://math.stackexchange.com/questions/12186/arc-length-of-b%C3%A9zier-curves
https://gamedev.stackexchange.com/questions/5373/moving-ships-between-two-planets-along-a-bezier-missing-some-equations-for-acce/5427#5427

Hoping this may help for implementation.

**Comment by MridulS** [comment]
I believe this was fixed in #7010
